### PR TITLE
Implement match group queries functions and test steps

### PR DESCRIPTION
**Pull request by dmikhalin**
## What is the goal of this PR?

We implement `match group` and `match group aggregate` queries functions and corresponding BDD test steps.

### CODE REVIEW

File: dependencies/vaticle/repositories.bzl [hunk 1]
```diff
@@ -38,8 +38,10 @@ def vaticle_typedb_common():
 def vaticle_typedb_protocol():
     git_repository(
         name = "vaticle_typedb_protocol",
-        remote = "https://github.com/vaticle/typedb-protocol",
-        commit = "aac5578529424bab662b7a66ca1e0419986ddc9d",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_protocol
+        remote = "https://github.com/dmitrii-ubskii/typedb-protocol",
+        branch = "api-rearchitecture",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_protocol
+#        remote = "https://github.com/vaticle/typedb-protocol",
+#        commit = "aac5578529424bab662b7a66ca1e0419986ddc9d",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_protocol
```
Comment (dmitrii-ubskii) [inline dependencies/vaticle/repositories.bzl:44]: Why? Is this a rebase error?

File: src/answer/concept_map_group.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    pub concept_maps: Vec<ConceptMap>,
+}
+
+// impl Clone for ConceptMapGroup {
+//     fn clone(&self) -> Self {
+//         let owner = self.owner.clone();
+//         let concept_maps = self.concept_maps.clone();
+//         Self { owner, concept_maps }
+//     }
+// }
```
Comment (dmitrii-ubskii) [inline src/answer/concept_map_group.rs:36]: Let's not leave this kind of stuff in.

File: src/answer/numeric_group.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// use crate::concept::Concept;
+// use crate::numeric::Numeric;
+
```
Comment (dmitrii-ubskii) [inline src/answer/numeric_group.rs:24]: Ditto

File: src/connection/network/proto/concept.rs [hunk 4]
```diff
@@ -74,6 +89,21 @@ impl TryFromProto<NumericProto> for Numeric {
...
 
+impl TryFromProto<ConceptMapGroupProto> for ConceptMapGroup {
+    fn try_from_proto(proto: ConceptMapGroupProto) -> Result<Self> {
+        let owner = match proto.owner {
+            Some(proto_owner) => Concept::try_from_proto(proto_owner)?,
+            None => return Err(ConnectionError::MissingResponseField("owner").into()),
+        };
+        let mut concept_maps = Vec::with_capacity(proto.concept_maps.len());
+        for cm in proto.concept_maps {
+            concept_maps.push(ConceptMap::try_from_proto(cm)?);
```
Comment (dmitrii-ubskii) [inline src/connection/network/proto/concept.rs:100]: `cm` => `concept_map`

**Review by dmitrii-ubskii — CHANGES_REQUESTED** [review]
Not much to say here, well done!

File: dependencies/vaticle/repositories.bzl (hunk 1 above)
Comment (dmikhalin) [reply dependencies/vaticle/repositories.bzl:44]: Might be.

File: src/connection/network/proto/concept.rs [hunk 5]
```diff
@@ -74,6 +85,19 @@ impl TryFromProto<NumericProto> for Numeric {
...
 }
 
+impl TryFromProto<ConceptMapGroupProto> for ConceptMapGroup {
+    fn try_from_proto(proto: ConceptMapGroupProto) -> Result<Self> {
+        let ConceptMapGroupProto { owner: owner_proto, concept_maps: concept_maps_proto } = proto;
+        let owner = Concept::try_from_proto(owner_proto.ok_or(ConnectionError::MissingResponseField("owner"))?)?;
+        let mut concept_maps = Vec::with_capacity(concept_maps_proto.len());
+        for concept_map in concept_maps_proto {
+            concept_maps.push(ConceptMap::try_from_proto(concept_map)?);
+        }
```
Comment (dmitrii-ubskii) [inline src/connection/network/proto/concept.rs:95]: You're doing manually what `.into_iter().map().try_collect()` can do for you.

**Review by dmitrii-ubskii — APPROVED** [review]
LGTM with a minor comment.

File: src/connection/network/proto/concept.rs [hunk 6]
```diff
@@ -63,9 +64,19 @@ impl IntoProto<AnnotationProto> for Annotation {
     }
 }
 
+impl TryFromProto<NumericGroupProto> for NumericGroup {
+    fn try_from_proto(proto: NumericGroupProto) -> Result<Self> {
+        let NumericGroupProto { owner: owner_proto, number: number_proto } = proto;
+        let owner = Concept::try_from_proto(owner_proto.ok_or(ConnectionError::MissingResponseField("owner"))?)?;
+        let numeric = Numeric::try_from_proto(number_proto.ok_or(ConnectionError::MissingResponseField("value"))?)?;
```
Comment (flyingsilverfin) [inline src/connection/network/proto/concept.rs:71]: The error field here should be `number` not `value` right?

**Review by flyingsilverfin — APPROVED** [review]
Looks good! One comment, please fix before merging

**Pull request merged**